### PR TITLE
Minor additional performance improvements to OrbitService

### PR DESCRIPTION
**Pull request by dpallotti**
#### Use double buffering instead of TracerThread::deferred_events_
This greatly reduced the number of times `std::vector::__emplace_back_slow_path`
is hit for this case (verified by capturing `OrbitService` with `Orbit`).

I've instrumented Trata's
`eva::graphics::rendering::spatial_geometry::world_sphere() const`
(the same as the YHITI tests) and observed the memory usage of `OrbitService`
after 30 s of capture, without noticing any significant difference.

Also switch to `absl::Mutex` as that's what we use in the majority of cases.
#### Double buffer instead of SenderThreadCaptureEventBuffer::event_buffer_
This greatly reduced the number of times `std::vector::__emplace_back_slow_path`
is hit for this case (verified by capturing `OrbitService` with `Orbit`).

I've instrumented Trata's
`eva::graphics::rendering::spatial_geometry::world_sphere() const`
(the same as the YHITI tests) and observed the memory usage of `OrbitService`
after 30 s of capture, without noticing any significant difference.
#### Further increase IDLE_TIME_ON_EMPTY_{RING_BUFFERS,DEFERRED_EVENTS}_US
We have seen in go/stadia-orbit-service-vs-perf that we still spend quite some
time calling `usleep`, so let's increase this further as the sizes of the
`perf_event_open` ring buffers allow for much longer times without being polled
(necessary if `OrbitService` is not scheduled for a while).

When capturing nothing with
```
./OrbitFakeClient --pid=10189 --frame_time=false --gpu_jobs=false \
--scheduling=false --sampling_rate=0
```
CPU utilization of OrbitService goes down from ~1.5% to ~0.5%.

This benefit is even higher on normal captures as the number of context switches
that end up being recorded is lower, as `OrbitService` itself causes fewer.

Test: Comparing with main when instrumenting Trata's `execute_task`: the
frequency of the (rare) lost/discarded events doesn't seem to increase.

### CODE REVIEW

File: src/Service/CaptureServiceImpl.cpp [hunk 1]
```diff
@@ -128,17 +134,17 @@ class GrpcCaptureEventSender final : public CaptureEventSender {
     LOG("Average number of bytes per event: %.2f", average_bytes);
   }
 
-  void SendEvents(std::vector<ClientCaptureEvent>&& events) override {
+  void SendEvents(std::vector<ClientCaptureEvent>* events) override {
     ORBIT_SCOPE_FUNCTION;
```
Comment (florian-kuebler) [inline src/Service/CaptureServiceImpl.cpp:138]: CHECK(events != nullptr);

File: src/Service/CaptureServiceImpl.cpp [hunk 2]
```diff
@@ -83,29 +84,34 @@ class SenderThreadCaptureEventBuffer final : public CaptureEventBuffer {
...
+      capture_event_sender_->SendEvents(&events_to_send_);
+      // std::vector::clear() "Leaves the capacity() of the vector unchanged", which is desired.
+      events_to_send_.clear();
     }
   }
 
-  std::vector<ClientCaptureEvent> event_buffer_;
-  absl::Mutex event_buffer_mutex_;
+  std::vector<ClientCaptureEvent> events_being_buffered_
+      ABSL_GUARDED_BY(events_being_buffered_mutex_);
```
Comment (florian-kuebler) [inline src/Service/CaptureServiceImpl.cpp:109]: Thanks for adding those!

**Review by florian-kuebler — APPROVED** [review]
LGTM, thanks!

File: src/Service/CaptureServiceImpl.cpp (hunk 1 above)
Comment (dpallotti) [reply src/Service/CaptureServiceImpl.cpp:138]: Done.

**Pull request merged**